### PR TITLE
Chem rework 7: a few researchchems and balance tweaks

### PR DESCRIPTION
**Pull request by jroinc**
## About The Pull Request

Slight changes to some chems, addition of four new chems for researchpeeps, and a defib change.

Toxins-

Neurotoxin effects in middle stage upped from 6 staminaloss to 8. Marines appear to have no issues handling it, and it's been unchanged since disarm was deleted.

Weedkiller buffed, should reliably kill weeds now.

Cyanide buffed. Should reliably kill now, still unobtainable.

Unstable mutagen changed, now it's poisonous.

Meds-
Water purgerate lowered from 1/tick to 0.2/tick. Everyone had a water bottle and was completely invalidating ayychems at no cost with a sip or two. Now, it takes longer.

Ryetalin toxin purgerate reduced to three from 5. Was too good.

Tricord adjusted to 0.5 brute and burnheal rather than 0.8. Toxheal lowered to 0.25 from 0.5. It now has 100% the healrate dedicated meds rather than 160%.

Larvaway toxin damage lowered, was too punishing.

Defib heart damage chance changed from 25% to 100%. It's 5 heart damage, and marines are pretty strong as-is. Maybe this will make them a bit less immortal.

Antitoxin no longer purges neuro at double-speed (or atomic coffee at all). Instead, it purges at normal speed.

Researchchems-

4 Researchchems added as of now, more may show up as I think of them. All researchchems require lemoline, which is req-purchasable or made from larval jelly. Larval jelly depreciated somewhat, recipes moved to lemoline to prevent duplication. Researcher spawns with a single 10u bottle of lemoline.

Quietus-

A chemical that causes rapid, large amounts of oxyloss upon a patient entering critical condition. It causes trace stamina damage upon initial injection for 2 minutes, warns the recipient once, then becomes latent. No OD given as generally having more of this is *worse* for you.

Under normal conditions, it only has a 5% chance/tick of doing one stamina damage. Upon the patient becoming unconscious for any reason, the chemical deals 25 oxyloss/tick, which should prevent bodydragging. Note that defibrilation sets the unconscious flag for a few seconds, more-or-less requiring dex+ or hypervene to allow a revive. This may require a rework to function properly, we'll see once it gets live testing.

If it's purged out, it deals 30 toxin damage upfront, and a message letting you know what happened so you know you're back in bodydrag meta.

Somolent-

A chemical that heals rapidly, as long as the player is unconscious or sleeping. It heals 0.2 points of brute and burn per tick, multiplied by cycle counter, capped at 10 of each via cycle counter. Post-cap it shuts off, as it should already have fully healed the target. No OD as the requirement to sleep is stringent enough. The chemical does not put players to sleep, but can and should be combined with chems that do.

A 2.5u inject heals 3 damage (of both types, for all these listed stats), a 5u inject heals 11 damage, a 7.5u inject heals 24 damage (and is now outhealing other, stacked chems), and a 10u inject 42 damage. A 15u inject heals a total of 93 damage of both types while it processes, and takes 25 ticks to do this. More than that heals progressively faster.

Medical nanites- 

These are designed to allow use of blood as a secondary health bar. Upon implantation of any amount, they cause a significant amount of damage over approximately five minutes. Once they settle down and are functional, they build up to 30u at 0.5u/tick, using 2 blood/tick to do so.

Upon taking either brute or burn damage while there are more than 5u of medical nanites, they heal 2 points per tick of each, using 1u/tick of nanites, for a total heal ratio of 1 point of damage healed per two units of blood. OD is at 36u, with 1 pt/tick of toxin damage, and should not be reachable normally.

~~Cryotox-~~

~~It gradually sets a person's bodytemp to 100 degrees, allowing cryogenic treatments to work. Nothing special here.~~
~~Would have been too OP. Now it sets stasis same as a bag for 10 seconds per unit initially injected.~~
Would have been incredibly annoying for almost no benefit, chem deleted. 4 it is.

Stimulon-

Oh boy. Increases movespeed by one while in the body. No OD threshold, and does not metabolize.

Passively increases by 0.2u/tick for every tick it's present in the body. Causes a random amount between 0.5 and 3 brute damage per tick. Also, causes a random amount between 0 and 0.05 genetic damage, multiplied by the current cycle. Purged by antitox.

## Why It's Good For The Game

Things for RSR to do. More chems with niche effects. Semi-annual med rebalance. Minimizes the number of immortal marines.

## Changelog
:cl:
add: Adds 4 new chems
balance: rebalances many chems listed in PR body, and defibs.
/:cl:

### CODE REVIEW

**Comment by jroinc** [comment]
This is a draft because I may add more rsrchems (and I have to let them start with some lemoline).

I also can't *actually test* this due to unknown byond errors, so I need to get that shit resolved.

**Comment by mathis650** [comment]
now we need the monkey cubes back to test those chems on them

**Comment by bowlofcereal** [comment]
god please give researcher something to do

File: code/modules/reagents/reagents/medical.dm [hunk 1]
```diff
@@ -1183,3 +1183,87 @@
...
+			if(prob(5))
+				to_chat(L, span_notice("You feel as though you should be sleeping for the medicine to work."))
+	return ..()
+
+
+/datum/reagent/medicine/research/medicalnanites
+	name = "Medical nanites"
+	description = "These are a batch of construction nanites altered for in-vivo replication. They can heal wounds using the iron present in the bloodstream. Medical care is recommended during injection."
+	color = "#19C832"
+	custom_metabolism = REAGENTS_METABOLISM * 0
```
Comment (QualityVan) [inline code/modules/reagents/reagents/medical.dm:1236]: ?

Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:1236]: What's the ? about?

Comment (QualityVan) [reply code/modules/reagents/reagents/medical.dm:1236]: Why not just 0?

File: code/modules/reagents/reagents/medical.dm [hunk 2]
```diff
@@ -1183,3 +1183,87 @@
...
+			if(prob(5))
+				to_chat(L, span_notice("You feel intense itching!"))
+		if(151 to INFINITY)
+			if(volume < 30) //smol injection will self-replicate up to 30u using 60u of blood.
+				L.reagents.add_reagent(/datum/reagent/medicine/research/medicalnanites, 0.5)
+				L.blood_volume -= 2
+				
+			var/brute_loss = L.getBruteLoss()
+			if(!brute_loss) //If we have no brute damage, cancel out
+				return ..()
```
Comment (QualityVan) [inline code/modules/reagents/reagents/medical.dm:1255]: This's gonna prevent burn healing whenever there's no brute damage.

Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:1255]: Early return or what?

Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:1255]: Or did I get the bruteheal/burnheal mixed up?

Comment (QualityVan) [reply code/modules/reagents/reagents/medical.dm:1255]: Yeah. If there's 0 brute loss you'll return out of on_mob_life before you reach the stuff about burn damage at all.

File: code/modules/reagents/reagents/medical.dm [hunk 3]
```diff
@@ -1183,3 +1183,87 @@
...
+			else if (volume > 5) //And if there's enough remaining, heal some damage and remove some nanites.
+				L.heal_limb_damage(2*effect_str, 0)
+				holder.remove_reagent(/datum/reagent/medicine/research/medicalnanites, 1)
+				
+			var/burn_loss = L.getFireLoss()
+			if(!burn_loss) //If we have no burn damage, cancel out
+				return ..()
+			else if (volume > 5) //And if there's enough remaining, heal some damage and remove some nanites.
```
Comment (QualityVan) [inline code/modules/reagents/reagents/medical.dm:1263]:
```suggestion
			if (volume > 5 && L.getFireLoss())
```
Same for brute.

File: code/modules/reagents/reagents/medical.dm [hunk 4]
```diff
@@ -1183,3 +1183,87 @@
...
+			L.adjustToxLoss(1*effect_str)
+			L.adjustStaminaLoss((1)*effect_str)
+			if(prob(5))
+				to_chat(L, span_notice("You feel intense itching!"))
+		if(151 to INFINITY)
+			if(volume < 30) //smol injection will self-replicate up to 30u using 60u of blood.
+				L.reagents.add_reagent(/datum/reagent/medicine/research/medicalnanites, 0.5)
+				L.blood_volume -= 2
+				
+			var/brute_loss = L.getBruteLoss()
```
Comment (QualityVan) [inline code/modules/reagents/reagents/medical.dm:1253]: This is gonna have issues with IB, which gives a positive on getBruteLoss but won't get helped by mob.heal_limb_damage.

File: code/modules/reagents/reagents/medical.dm [hunk 5]
```diff
@@ -1183,3 +1183,87 @@
 			continue
 		limb_to_unfix.fracture()
 		break
+
+
+/datum/reagent/medicine/research
```
Comment (QualityVan) [inline code/modules/reagents/reagents/medical.dm:1188]: Is having a parent type purely for organisation, or are you planning something else with it?

File: code/modules/reagents/reagents/medical.dm [hunk 6]
```diff
@@ -1183,3 +1183,87 @@
...
+
+/datum/reagent/medicine/research/somolent
+	name = "Somolent"
+	description = "This is a highly potent regenerative drug, designed to heal critically injured personnel. Only functions on unconscious or sleeping people."
+	color = "#19C832"
+	custom_metabolism = REAGENTS_METABOLISM * 2.5 //0.5u/ticc
+	taste_description = "naptime"
+
+/datum/reagent/medicine/research/somolent/on_mob_life(mob/living/L, metabolism)
+	switch(current_cycle)
```
Comment (QualityVan) [inline code/modules/reagents/reagents/medical.dm:1223]: Switch statement for a single case?

File: code/modules/reagents/reactions/medical.dm [hunk 7]
```diff
@@ -15,12 +15,6 @@
 	required_reagents = list(/datum/reagent/oxygen = 2, /datum/reagent/toxin/phoron = 0.1)
 	required_catalysts = list(/datum/reagent/toxin/phoron = 5)
 
-/datum/chemical_reaction/dermaline
-	name = "Dermaline"
-	results = list(/datum/reagent/medicine/dermaline = 3)
-	required_reagents = list(/datum/reagent/oxygen = 1, /datum/reagent/phosphorus = 1, /datum/reagent/medicine/kelotane = 1)
-	required_catalysts = list(/datum/reagent/consumable/larvajelly = 5)
```
Comment (QualityVan) [inline code/modules/reagents/reactions/medical.dm:22]: The jelly recipes don't consumes the jelly, the limoline ones do. Means each larva you grind is only gonna give you 15u of mera or derm. Intended?

File: code/modules/reagents/reagents/medical.dm (hunk 4 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:1253]: Intentional, and good catch.

File: code/modules/reagents/reagents/medical.dm (hunk 5 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:1188]: Both, I might do something with them. I'm considering just leaving them as reagent/research rather than medicine/research, but I'm still thinking it through.

File: code/modules/reagents/reagents/medical.dm (hunk 6 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:1223]: Because I want it to end after X... wait, can I just leave it as current_cycle? Self-taught here, and I've only seen switch(current_cycle) before.

File: code/modules/reagents/reactions/medical.dm (hunk 7 above)
Comment (jroinc) [reply code/modules/reagents/reactions/medical.dm:22]: Intended. I'd like to keep both, but the duplicate recipes are an issue and we never get larba. Remind me in a few and I'll look at req efficiencies, as larva are uncommon now.

File: code/modules/reagents/reagents/medical.dm (hunk 3 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:1263]: ah yes

File: code/modules/reagents/reagents/medical.dm (hunk 6 above)
Comment (QualityVan) [reply code/modules/reagents/reagents/medical.dm:1223]: Just if(current_cycle < 51), yeah.

Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:1223]: So that would just be-
on mob life
*everything it does*
then *if current cycle* followed by nothing?

File: code/modules/reagents/reagents/medical.dm (hunk 1 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:1236]: oh yeah, good point :P 

I'll change it next edit for both chems

**Comment by jroinc** [comment]
Phase one testing complete- functions as intended. Minor changes made to chems for usability purposes.

**Comment by jroinc** [comment]
Changes made, testing complete. Everything's functioning as intended, ready for review, TMC, and live testing eventually.

**Comment by EvilMelonTheSecond** [comment]
The defib change seems a bit dangerous since heart damage can permakill marines and there is no way to fix the organs of a corpse as far as I know. Corpsman don't always know the mechanics of the game 100% and this just seems like a pretty brutal pitfall for something they are likely gonna do many times in a round.

**Comment by QualityVan** [comment]
> 
> 
> The defib change seems a bit dangerous since heart damage can permakill marines and there is no way to fix the organs of a corpse as far as I know. Corpsman don't always know the mechanics of the game 100% and this just seems like a pretty brutal pitfall for something they are likely gonna do many times in a round.

You can fix up corpse organs via surgery.

File: code/modules/reagents/reagents/medical.dm [hunk 8]
```diff
@@ -1183,3 +1183,99 @@
...
+	scannable = TRUE
+	taste_description = "naptime"
+
+/datum/reagent/medicine/research/somolent/on_mob_life(mob/living/L, metabolism)
+	if(current_cycle > 50)
+		return ..()
+	if(L.stat == UNCONSCIOUS)
+		L.heal_limb_damage(0.2*current_cycle*effect_str, 0.2*current_cycle*effect_str)
+		if(prob(20))
+			to_chat(L, span_notice("You feel as though you should be sleeping for the medicine to work."))
```
Comment (BraveMole) [inline code/modules/reagents/reagents/medical.dm:1239]: but you're already sleeping..?

**Review by BraveMole — CHANGES_REQUESTED** [review]
This should be atomised.

I don't see why defib changes are in this, especially since you are not adding anything to help with heart damage?

Same with the water one, it's just a straight rebalance, doesn't belong here

**Comment by jroinc** [comment]
> 
> 
> This should be atomised.
> 
> I don't see why defib changes are in this, especially since you are not adding anything to help with heart damage?
> 
> Same with the water one, it's just a straight rebalance, doesn't belong here

Because I already added the stuff to help with heart damage in a previous PR.

**Comment by jroinc** [comment]
> 
> 
> The defib change seems a bit dangerous since heart damage can permakill marines and there is no way to fix the organs of a corpse as far as I know. Corpsman don't always know the mechanics of the game 100% and this just seems like a pretty brutal pitfall for something they are likely gonna do many times in a round.

You can fix the organs of a corpse. And yes, I want this to permakill marines that require more than six defibs at once. Medevac them.

File: code/modules/reagents/reagents/medical.dm (hunk 8 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:1239]: Do you know of a message category that only functions while awake?

**Comment by jroinc** [comment]
forgot I wanted to do some more stuff, be back in a bit

**Comment by jroinc** [comment]
Aight. It compiles. Time to test everything again.

**Comment by jroinc** [comment]
K, everything works as intended except cryotox. I'll go fix it in a few hours. 

Also, weedkiller smoke is invisible. I didn't change anything that could cause that so ???

**Comment by jroinc** [comment]
Cryotox delet. I figured the issue with the first one, but portable cryo anywhere is far too OP (injectors that insta-fix anything but shrapnel and larva). My second idea was a stasis chem, but that was just a bad idea as it's a one-click stun for minutes.

So, 4 researchchems it is.

**Comment by jroinc** [comment]
I'll get back to this (and all the other PR's I have) in a while. Stuff going on I have to deal with first.

**Comment by davidchan13** [comment]
Really, really don't agree with the Ryetalyn change, it's purged by 3 commonly used marine meds, and only has one function. Dylo and Water changes are very 'eh' and feel like kneejerk reactions.

**Comment by QualityVan** [comment]
What about tying water's purge rate to cycle_count, rather than just cutting the base amount?

**Comment by SplinterGP** [comment]
image
